### PR TITLE
Configure build scan plugin

### PR DESCRIPTION
**Pull request by aalmiray**
This small PR configures the `build-scan` Gradle plugin (https://guides.gradle.org/creating-build-scans/) which yields further insight into the build, allowing the team to make informed decisions on where and when the build may be tweaked to get faster, more performant builds.

Here's a sample scan https://scans.gradle.com/s/heq3wy556se7w/ of my local fork.

### CODE REVIEW

**Comment by jponge** [comment]
This binds to an agreement with Gradle Inc., right?

**Comment by aalmiray** [comment]
It does. I can send an update to make the agreement conditionally for each developer.

**Comment by jponge** [comment]
I think he’d be best if that plugin was conditional.

**Comment by jponge** [comment]
Thanks @aalmiray 

Would it be possible to have the script in Kotlin rather than Groovy?

**Comment by aalmiray** [comment]
Sure. My kotlin-fu is not strong enough but I can give it a try

File: build.gradle.kts [hunk 1]
```diff
@@ -19,12 +19,55 @@ import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
...
 
 plugins {
+  id("com.gradle.build-scan") version "2.2.1"
   `java-gradle-plugin`
   kotlin("jvm") version "1.3.21"
   id("com.github.ben-manes.versions") version "0.21.0"
   id("com.gradle.plugin-publish") version "0.10.1"
 }
 
+fun str2bool(s: String?, v: String = "true"): Boolean {
```
Comment (jponge) [inline build.gradle.kts]: I'd have an extension method here, and at least I'd simplify with `return "true" == s` as Kotlin has an equality check operator that makes the _right_ thing.

**Comment by aalmiray** [comment]
Closing due to inactivity. Feel free to reach out to Gradle's Dev Advocacy should you like to continue.

**Pull request closed without merging**